crates/steel-parser/src/lexer.rs

```diff
@@ -271,9 +271,9 @@ impl<'a> Lexer<'a> {
     fn read_word(&mut self) -> TokenType<&'a str> {
         while let Some(&c) = self.chars.peek() {
             match c {
-                '(' | '[' | ')' | ']' => break,
+                '(' | '[' | ')' | ']' | '{' | '}' => break,
                 c if c.is_whitespace() => break,
-                '\'' => {
+                '\'' | '"' | '`' | ';' | ',' => {
```

Comment:
This is the original rabbit hole I was getting into: improving how the reader handles identifiers, in preparation for supporting escaped identifiers `|...|`. Alas, more work for the future.

Reply:
So are `{` and `}` no longer legal in identifiers?

Reply:
They're not, which is also in line with Racket.